compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -149,10 +149,8 @@ pub fn unsized_info<'tcx, Cx: CodegenMethods<'tcx>>(
         }
         (_, &ty::Dynamic(ref data, ..)) => {
             let vtable_ptr = cx.layout_of(cx.tcx().mk_mut_ptr(target)).field(cx, FAT_PTR_EXTRA);
-            cx.const_ptrcast(
-                meth::get_vtable(cx, source, data.principal()),
-                cx.backend_type(vtable_ptr),
-            )
+            let vtable_value = meth::get_vtable(cx, source, data.principal());
+            cx.const_ptrcast(vtable_value, cx.backend_type(vtable_ptr))
```

Comment:
This change doesn't seem to be necessary (anymore). Can you revert it?
```suggestion
            cx.const_ptrcast(
                meth::get_vtable(cx, source, data.principal()),
                cx.backend_type(vtable_ptr),
            )
```

Reply:
Indeed!